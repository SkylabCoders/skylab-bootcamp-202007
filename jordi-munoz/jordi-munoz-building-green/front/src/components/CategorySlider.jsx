import React, { useState } from 'react';
import { makeStyles, useTheme } from '@material-ui/core/styles';
import MobileStepper from '@material-ui/core/MobileStepper';
import Button from '@material-ui/core/Button';
import KeyboardArrowLeft from '@material-ui/icons/KeyboardArrowLeft';
import KeyboardArrowRight from '@material-ui/icons/KeyboardArrowRight';
import Form from './Form';

const useStyles = makeStyles({
  root: {
    maxWidth: 400,
    flexGrow: 1,
  },
});

export default function CategorySlider() {
  const classes = useStyles();
  const theme = useTheme();
  const [activeStep, setActiveStep] = useState(0);

  const handleNext = () => {
    setActiveStep((prevActiveStep) => prevActiveStep + 1);

  };

  const handleBack = () => {
    setActiveStep((prevActiveStep) => prevActiveStep - 1);
  };

  return (
    <div style={{
      marginTop: "80px",
      marginBottom: "60px",
      display: "flex",
      flexDirection: "column"
    }}>
      <MobileStepper
        variant="progress"
        steps={3}
        position="static"
        activeStep={activeStep}
        className={classes.root}
        nextButton={
          <Button size="small" onClick={handleNext} disabled={activeStep === 2}>
            Next
          {theme.direction === 'rtl' ? <KeyboardArrowLeft /> : <KeyboardArrowRight />}
          </Button>
        }
        backButton={
          <Button size="small" onClick={handleBack} disabled={activeStep === 0}>
            {theme.direction === 'rtl' ? <KeyboardArrowRight /> : <KeyboardArrowLeft />}
          Back
        </Button>
        }
      />
      <Form step={activeStep} stepChange={setActiveStep} />
    </div>
  );
}