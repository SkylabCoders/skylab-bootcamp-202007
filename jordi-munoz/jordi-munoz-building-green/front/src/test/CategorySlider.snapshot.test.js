import renderer from 'react-test-renderer';
import React from 'react';
import CategorySlider from '../components/CategorySlider';
import { BrowserRouter } from 'react-router-dom';

describe('CategorySlider snapshot', () => {
  const categorySliderTree = renderer.create(
    <BrowserRouter>
      <CategorySlider />
    </BrowserRouter>
  );
  it('should match', () => {
    expect(categorySliderTree.toJSON()).toMatchSnapshot();
  });

});