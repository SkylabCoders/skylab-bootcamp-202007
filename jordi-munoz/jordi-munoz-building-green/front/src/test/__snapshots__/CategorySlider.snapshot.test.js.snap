// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategorySlider snapshot should match 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "marginBottom": "60px",
      "marginTop": "80px",
    }
  }
>
  <div
    className="MuiPaper-root MuiMobileStepper-root MuiMobileStepper-positionStatic makeStyles-root-1 MuiPaper-elevation0"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
          />
        </svg>
        Back
      </span>
    </button>
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={0}
      className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiMobileStepper-progress MuiLinearProgress-determinate"
      role="progressbar"
    >
      <div
        className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
        style={
          Object {
            "transform": "translateX(-100%)",
          }
        }
      />
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Next
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
          />
        </svg>
      </span>
    </button>
  </div>
  <div>
    Loading...
  </div>
</div>
`;
